tests/saitests/py3/sai_qos_tests.py

```diff
@@ -2854,31 +2866,45 @@ def runTest(self):
             else:
                 send_packet(
                     self, self.src_port_ids[sidx], pkt, self.pkts_num_leak_out)
-
+            time.sleep(1)
             # send packets to all pgs to fill the service pool
             # and trigger PFC on all pgs
             for i in range(0, self.pgs_num):
-                # Prepare TCP packet data
-                tos = sidx_dscp_pg_tuples[i][1] << 2
-                tos |= self.ecn
-                ttl = 64
-                default_packet_length = self.pkt_size
-                pkt = simple_tcp_packet(pktlen=default_packet_length,
-                                        eth_dst=self.router_mac if self.router_mac != '' else self.dst_port_mac,
-                                        eth_src=self.src_port_macs[sidx_dscp_pg_tuples[i][0]],
-                                        ip_src=self.src_port_ips[sidx_dscp_pg_tuples[i][0]],
-                                        ip_dst=self.dst_port_ip,
-                                        ip_tos=tos,
-                                        ip_ttl=ttl)
+                # Prepare IP packet data
+                pkt = construct_ip_pkt(self.pkt_size,
+                                       self.pkt_dst_mac,
+                                       self.src_port_macs[sidx_dscp_pg_tuples[i][0]],
+                                       self.src_port_ips[sidx_dscp_pg_tuples[i][0]],
+                                       self.dst_port_ip,
+                                       sidx_dscp_pg_tuples[i][1],
+                                       self.src_port_vlan[sidx_dscp_pg_tuples[i][0]],
+                                       ecn=self.ecn,
+                                       ttl=64)
```

Comment:
How are we setting TOS byte here ?

Reply:
As per the definition it is set as below 
def construct_ip_pkt(pkt_len, dst_mac, src_mac, src_ip, dst_ip, dscp, src_vlan, **kwargs):
    ecn = kwargs.get('ecn', 1)
    ip_id = kwargs.get('ip_id', None)
    ttl = kwargs.get('ttl', None)
    exp_pkt = kwargs.get('exp_pkt', False)

    **tos = (dscp << 2) | ecn**
    pkt_args = {
        'pktlen': pkt_len,
        'eth_dst': dst_mac,
        'eth_src': src_mac,
        'ip_src': src_ip,
        'ip_dst': dst_ip,
        'ip_tos': tos
    }